File: parse_kml.py
import xml.etree.ElementTree as ET

tree = ET.parse('C:/Users/hyder/Desktop/compassion-course-website/compassion_map.kml')
root = tree.getroot()

ns = {'kml': 'http://www.opengis.net/kml/2.2'}

placemarks = root.findall('.//kml:Placemark', ns)
print(f'Total placemarks: {len(placemarks)}')

coords = []
for pm in placemarks:
    name_el = pm.find('kml:name', ns)
    name = name_el.text.strip() if name_el is not None and name_el.text else 'Unknown'

    coord_el = pm.find('.//kml:coordinates', ns)
    if coord_el is not None and coord_el.text:
        parts = coord_el.text.strip().split(',')
        if len(parts) >= 2:
            try:
                lon = float(parts[0])
                lat = float(parts[1])
                coords.append((lat, lon, name))
            except ValueError:
                pass

print(f'Total coordinates extracted: {len(coords)}')

# Write all coordinates to output file
with open('C:/Users/hyder/Desktop/compassion-course-website/all_coordinates.txt', 'w', encoding='utf-8') as f:
    for lat, lon, name in coords:
        f.write(f'[{lat}, {lon}], // {name}\n')

print('Written to all_coordinates.txt')

File: parse_kml.js
const fs = require('fs');

const kml = fs.readFileSync('C:/Users/hyder/Desktop/compassion-course-website/compassion_map.kml', 'utf-8');

// Extract placemarks with name and coordinates
const placemarkRegex = /<Placemark>\s*<name>([^<]*)<\/name>[\s\S]*?<coordinates>\s*([\d\-.,\s]+)\s*<\/coordinates>[\s\S]*?<\/Placemark>/g;

const coords = [];
const seen = new Set();
let match;

while ((match = placemarkRegex.exec(kml)) !== null) {
  const name = match[1].trim();
  const coordStr = match[2].trim();
  const parts = coordStr.split(',');
  if (parts.length >= 2) {
    const lon = parseFloat(parts[0]);
    const lat = parseFloat(parts[1]);
    if (!isNaN(lat) && !isNaN(lon)) {
      const key = `${lat.toFixed(6)},${lon.toFixed(6)}`;
      if (!seen.has(key)) {
        seen.add(key);
        coords.push({ lat, lon, name });
      }
    }
  }
}

console.log(`Total unique coordinates: ${coords.length}`);

// Write to file
const lines = coords.map(c => `[${c.lat}, ${c.lon}], // ${c.name}`);
fs.writeFileSync('C:/Users/hyder/Desktop/compassion-course-website/all_coordinates.txt', lines.join('\n'), 'utf-8');
console.log('Written to all_coordinates.txt');
